IbisHardwareIGSIO/ibishardwareIGSIO.cpp

```diff
@@ -392,38 +392,20 @@ void IbisHardwareIGSIO::InitPlugin()
 
     // Look for the Plus Toolkit path and Plus Server executable
     m_plusServerExec = GetIbisAPI()->GetCustomPath( PlusServerExecutable );
-    bool ok = true;
+    bool ok = false;
     if( m_plusServerExec != QString::null && !m_plusServerExec.isEmpty() )
     {
         QFileInfo fi( m_plusServerExec );
         if( !( fi.isFile() && fi.isExecutable() ) )
-            ok = false;
-    }
-    else
-    {
-        m_plusToolkitPathsFilename = m_plusConfigDirectory + QString("plus-toolkit-paths.txt");
-        if( QFile::exists( m_plusToolkitPathsFilename ) )
         {
-            QFile pathFile( m_plusToolkitPathsFilename );
-            pathFile.open( QIODevice::ReadOnly | QIODevice::Text );
-            QTextStream pathIn( &pathFile );
-            pathIn >> m_plusServerExec;
-            QFileInfo fi( m_plusServerExec );
-            if( fi.isFile() && fi.isExecutable() )
-                GetIbisAPI()->RegisterCustomPath( PlusServerExecutable, fi.absoluteFilePath() );
-            else
-                ok = false;
+            GetIbisAPI()->UnRegisterCustomPath( PlusServerExecutable );
         }
-        else ok = false;
+        else
+            ok = true;
     }
     if( !ok )
     {
-        GetIbisAPI()->UnRegisterCustomPath( PlusServerExecutable );
         GetIbisAPI()->RegisterCustomPath( PlusServerExecutable, "" );
```

Comment:
We should always register the path if it is not part of the list of custom paths. We don't care if it is valid or not, it needs to be registered in order to appear in the prefs dialog and allow the user to set it. 

So you should simply the following in the beginning of the function:
```
if( GetIbisAPI()->GetCustomPath( PlusServerExecutable ) == QString::null)
    GetIbisAPI()->RegisterCustomPath( PlusServerExecutable );
```
Or even better, modify the function RegisterCustomPath so that it only registers the path if it doesn't exist and then in the plugin, you don't bother and you simply register the path every time

Reply:
I still have to check the validity of the path in the plugin, if Register Path does not register the existing path, it can be anything

Reply:
Well, this piece of code is useless because it is replicated in LaunchLocalServer. In InitPlugin, I would simply make sure the path is registered.

Reply:
This code makes sure, that the path is registered. In Launch... I still have to check the validity. 
Maybe I do not know enough about launching server. Lets say, I start ibis, the server is found, but then, I open preferences, change PlusServer and try to launch it  - is it allowed?
Then, I have to validate the PlusServer variable and reset it if the new one given by the user is invalid. Also user may simply scrap the variable.

Reply:
If I have to make sure, that the server is registered in InitPlugin, I still have to use this code as I don't have any default. Of course, all this can be done once only in Launch...  leaving the server undefined until then if it was not set in preferences.

Reply:
The server isn't and shouldn't be started anywhere else than in LauchLocalServer, so it is meaningless to check the validity of the executable elsewhere, it only creates code duplication. The HardwareModule will call LauchLocalServer automatically if specified in the config file, but the server can also be launched by selecting a new configuration while Ibis is running.  There are 2 typical cases where you will want to change the path of the server executable: if it has never been specified or if PlusServer has been moved or changed location. In both cases:

1. The user will launch Ibis
2. A hardware config file will cause LaunchLocalServer to be called, which will generate an error.
3. The user will open the preference window and set the proper path to PlusServer
4. The user will go back to the hardware config window and launch the config again. This time it will succeed.

Reply:
> If I have to make sure, that the server is registered in InitPlugin, I still have to use this code as I don't have any default. Of course, all this can be done once only in Launch... leaving the server undefined until then if it was not set in preferences.

Of course, you have to register the path in InitPlugin(). My point is you shouldn't check validity here.

Reply:
b.t.w. The preference class should make sure that you don't register 2 paths with the same name. This way, the plugins wouldn't have to bother with that and would simply register their path systematically in InitPlugin().